FaultToleranceDefinitionException after migrating to Quarkus 2.0

## Describe the bug
After migrating to Quarkus 2.0, @Retry throws the following exception:
```
Caused by: javax.enterprise.inject.spi.DeploymentException: org.eclipse.microprofile.faulttolerance.exceptions.
FaultToleranceDefinitionException: Invalid @Blocking/@NonBlocking on public java.lang.String org.acme.getting.
started.ReactiveGreetingResource.greeting(java.lang.String): must return java.util.concurrent.CompletionStage or io.smallrye.mutiny.Uni
```
Using resteasy-reactive extension, the problem takes place on blocking endpoints with non Uni or CompletionStage return values.
The problem can be mitigated by:
 1. removing @Retry annotation
 2. rolling back to pre Quarkus 2.0 versions
 3. moving from resteasy-reactive to non-reactive resteasy.

### Expected behavior
Not getting FaultToleranceDefinitionException.

### Actual behavior
FaultToleranceDefinitionException thrown.

## To Reproduce
https://github.com/jjolsvai/quarkus-ft-reproducer

Steps to reproduce the behavior:
1. run the unit test in re-producer

## Additional context
We have a common library with resteasy reactive, but not all the microservices moved to full reactive yet. Hence, there are blocking endpoints.

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -48,7 +48,7 @@
         <smallrye-open-api.version>2.1.7</smallrye-open-api.version>
         <smallrye-graphql.version>1.2.8</smallrye-graphql.version>
         <smallrye-opentracing.version>2.0.1</smallrye-opentracing.version>
-        <smallrye-fault-tolerance.version>5.2.0</smallrye-fault-tolerance.version>
+        <smallrye-fault-tolerance.version>5.2.1</smallrye-fault-tolerance.version>
         <smallrye-jwt.version>3.2.0</smallrye-jwt.version>
         <smallrye-context-propagation.version>1.2.0</smallrye-context-propagation.version>
         <smallrye-reactive-streams-operators.version>1.0.13</smallrye-reactive-streams-operators.version>

